src/architecture/storage.md

```diff
@@ -1,301 +1,40 @@
 # Exonum Data Model
 
-This page describes Exonum **data storage** principles, from the database engine
-used (RocksDB), to the abstractions that are used in client
-applications.
+This page provides an overview of how Exonum works with the persistent data.
 
-1. [Exonum table types](#table-types) lists supported types of
-   data storage collections. Tables represent the highest abstraction level
-   for data storage
-2. [Low-level storage](#low-level-storage) explains how tables are persisted
-   using RocksDB
-3. [View layer](#view-layer) describes the wrapper over the DB engine
-   that ensures atomicity of blocks and transactions
-4. [List of system tables](#system-tables) contains tables
-   used directly by the Exonum core
-5. [Indexing](#indexing) gives an insight how indices over structured data
-   can be built in Exonum
+Exonum uses MerkleDB as a storage framework. MerkleDB is an object database. It
+represents a high-level wrapper over the key-value store.
 
-## Table Types
+Currently [RocksDB][rocks-db] is used for low-level key-value storage engine in
+Exonum.
 
-Tables (aka indices) perform the same role as in relational database
-management systems (RDBMSs). Every
-table stores records of a specific type. However, unlike RDBMS tables,
-all Exonum tables internally are implemented as wrappers around key-value
-stores.
-Both keys and values in the wrapped stores are persisted as byte sequences.
-Exonum does not natively support operations (matching, grouping, sorting, etc.)
-over separate value fields, as it is the case with other key-value storages.
+The objects of MerkleDB are convenient abstractions for work with
+blockchain-specific data structures. For example, such abstraction can be a
+list that provides cryptographic proofs for its stored items.
 
-### Key Sorting and Iterators
+Currently, the objects in MerkleDB fall into two types:
 
-Exonum tables implement iterators over stored items (or keys, values, and
-key-value
-pairs in the case of maps). Such
-iterators use key ordering of the underlying key-value storage to determine
-the iteration order.
-Namely, keys are lexicographically ordered over their binary serializations;
-this ordering coincides with that used in
-[RocksDB](#low-level-storage).
+- **blobs**, which represent sequences of bytes, and
+- **root objects**, that do not have parents. These objects have UTF-8
```

Comment:
Nit: shouldn't it be "UTF-8 string identifiers"?

Reply:
edited